File: Flutter App/unicoVehicle/lib/Models/Vehicle.dart
import './FuelType.dart';
import './CylinderArrangement.dart';
import './VehicleName.dart';
import './VehicleType.dart';
import './TransmissionType.dart';
import './VehicleVariant.dart';
import './VehicleFeatures.dart';

class Vehicle {
  final int vehicleId;
  final VehicleVariant vehicleVariant;
  final TransmissionType transmissionType;
  final FuelType fuelType;
  final VehicleType vehicleType;
  final CylinderArrangement cylinderArrangement;
  final VehicleName vehicleName;
  final int doors;
  final int passenger;
  final int cylinder;
  final int engineSize;
  final double horsePower;
  final double torque;
  final double mileage;
  final double height;
  final double width;
  final double length;
  final double groundClearance;
  final double wheelBase;
  final double fuelTankSize;
  final double kerbWeight;
  final double grossWeight;
  final VehicleFeatures vehicleFeatures;

  Vehicle({
    this.cylinder,
    this.cylinderArrangement,
    this.doors,
    this.engineSize,
    this.fuelTankSize,
    this.fuelType,
    this.grossWeight,
    this.groundClearance,
    this.height,
    this.horsePower,
    this.kerbWeight,
    this.length,
    this.mileage,
    this.passenger,
    this.torque,
    this.transmissionType,
    this.vehicleFeatures,
    this.vehicleId,
    this.vehicleName,
    this.vehicleType,
    this.vehicleVariant,
    this.wheelBase,
    this.width,
  });
}

File: UnicoVehicle/UnicoVehicle/UnicoVehicle.DTO/Vehicle.cs
using System;
namespace UnicoVehicle.DTO
{
    public class Vehicle
    {
        public int VehicleId { get; set; }
        public VehicleVariant VehicleVariant { get; set; }
        public TransmissionType TransmissionType { get; set; }
        public FuelType FuelType { get; set; }
        public VehicleType VehicleType { get; set; }
        public CylinderArrangement CylinderArrangement { get; set; }
        public VehicleName VehicleName { get; set; }
        public int Doors { get; set; }
        public int Passenger { get; set; }
        public int Cylinder { get; set; }
        public int EngineSize { get; set; }
        public double HorsePower { get; set; }
        public double Torque { get; set; }
        public double Mileage { get; set; }
        public double Height { get; set; }
        public double Width { get; set; }
        public double Length { get; set; }
        public double GroundClearance { get; set; }
        public double WheelBase { get; set; }
        public double FuelTankSize { get; set; }
        public double KerbWeight { get; set; }
        public double GrossWeight { get; set; }
        public VehicleFeatures VehicleFeatures { get; set; }
    }
}
